File: 326/power-of-three.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfThree = function (n) {
    if (n == 0)
        return false;
    let a = Math.log(n) / Math.log(3);
    return Math.pow(3, Math.round(a)) === n;
};

File: 326/power-of-three.go
package main

import "fmt"

func isPowerOfThree(n int) bool {
	if n < 1 {
		return false
	}

	for n%3 == 0 {
		n /= 3
	}

	return n == 1
}

func main() {
	n := 27
	fmt.Println(isPowerOfThree(n))
}
